### PR TITLE
Creation Of Classes For The School Library

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR We present Creation Of Classes For The School Library🏫 :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- Setup ruby workflow and linters 🔍 
- use GitFlow 🌊.
- Create a classes `Person`, `Student` & `Teacher` that have an instance variables 📎  `@id`,`@name`,`@age`,`@parent_permission` as main instance variables.
- Create subClass instance Variables 🏛️  : `@classroom` & `@specialization`.
- Create `methods` for all these classes. 🎣 

## Achievements
- Setup ruby workflow and linters 🔍 ✅
- use GitFlow 🌊.✅
- Create a classes `Person`, `Student` & `Teacher` that have an instance variables 📎  `@id`,`@name`,`@age`,`@parent_permission` as main instance variables.✅
- Create subClass instance Variables 🏛️  : `@classroom` & `@specialization`.✅
- Create `methods` for all these classes. 🎣 ✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  private
+
+  def of_age?
+    @age > 18
+  end
+
+  def can_use_services?
+    @age > 18 || @parent_permission
+  end
```
Comment (OyePriscilla) [inline person.rb]: - [ ] Please kindly take note that these two methods are marked as private, which means they can only be called from within the class itself and cannot be accessed directly from outside the class. This opposed the project requirement, indly ensure that `can_use_services?` is marked as public. 👍 

```suggestion
def can_use_services?
    @age > 18 || @parent_permission
  end
  
  private

  def of_age?
    @age > 18
  end  
```

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @gisachris 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
**Implemented classes and objects in Ruby** ✔️ 
**Implemented encapsulation and inheritance with Ruby** ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (gisachris) [reply person.rb]: Thank you @OyePriscilla for the great insight and review 👍 I will make the required changes.

**Comment by gisachris** [comment]
## IN This PR We present Requested changes ♻️ on Creation Of Classes For The School Library🏫 :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Required Changes ♻️
- Make the `can_use_service?` method a public method🎣  
> The above was requested [here](https://github.com/gisachris/school-library-app/pull/1#discussion_r1279542610)



## Achievements
- Make the `can_use_service?` method a public method🎣 ✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @gisachris , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**